src/main/java/com/flowingcode/vaadin/addons/cronexpressioneditor/CronExpressionEditor.java

```diff
@@ -0,0 +1,582 @@
+    if (days.size() == 7 || days.size() == 0) {
+      values.put(5, "*");
+    } else {
+      values.put(5, String.join(",", aux));
+    }
+    values.put(3, "?");
+    updateCronExpression(values);
+  }
+
+  private void getDailyExpression(Units timeUnit, int value) {
+    Map<Integer, String> values = new HashMap<>();
+
+    String aux = "*/" + String.valueOf(value);
+    if (inputExpressionTf.getValue().isEmpty()) {
+      updateCronExpression(DEFAULT_CRON);
+    }
+    switch (timeUnit) {
+      case SECONDS:
+        values.put(0, aux);
+        values.put(1, "0");
+        values.put(2, "0");
+        break;
+      case MINUTES:
+        values.put(0, "0");
+        values.put(1, aux);
+        values.put(2, "0");
+        break;
+      case HOURS:
+        values.put(0, "0");
+        values.put(1, "0");
+        values.put(2, aux);
+        break;
+    }
+    updateCronExpression(values);
+  }
+
+  private void showNextDates() {
+    if (!inputExpressionTf.getValue().isEmpty()) {
+      grid.removeAllColumns();
+      grid.addColumn(date -> formatDate(date));
+      grid.setHeight("30%");
+      grid.setItems(DataProvider.fromCallbacks(q -> generateNextDate(LocalDateTime.now(), q.getOffset(), q.getLimit()),
+          q -> Integer.MAX_VALUE));
+      add(grid);
+    }
+  }
+
+  private LocalDateTime calculateNextDate(LocalDateTime date) {
+    return CronExpression.parse(inputExpressionTf.getValue()).next(date);
+  }
+
+  private Stream<LocalDateTime> generateNextDate(LocalDateTime start, int offset, int limit) {
+    Stream<LocalDateTime> nextDates = Stream.iterate(calculateNextDate(start), this::calculateNextDate);
+    return nextDates.skip(offset).limit(limit);
+  }
+
+  private String formatDate(LocalDateTime date) {
+    DateTimeFormatter dateFormatter = DateTimeFormatter.ofLocalizedDateTime(FormatStyle.LONG, FormatStyle.MEDIUM)
+        .withLocale(locale == null ? Locale.ENGLISH : locale);
+    return date.format(dateFormatter);
+  }
+
+  /**
+   * Enables or disables user input for the cron expression text field.
+   * 
+   * @param isEnabled a {@code boolean} to set the editing mode of the cron expression TextField.
+   */
+  public void setCronInputEnabled(boolean isEnabled) {
+    cronInputEnabled = isEnabled;
+    inputExpressionTf.setReadOnly(!isEnabled);
+  }
+
+  /**
+   * Displays a button that generates a grid with the next dates scheduled by the cron expression generated
+   * 
+   * @param nextDatesVisible a {@code boolean} to set the visibility of the button.
+   */
+  public void setNextDatesButtonVisible(boolean nextDatesVisible) {
+    nextDatesBtn.setVisible(nextDatesVisible);
+    if (!nextDatesBtn.isVisible()) {
+      remove(grid);
+    }
+  }
+
+  /**
+   * Allows the developer to set a default expression.
+   * 
+   * @param cron: a {@code String} representing the expression to be set as default.
+   * @return a {@code boolean} indicating if the expression is valid or not.
```

Comment:
Please remove ":" after cron and description for param and return should not end with period. See https://github.com/FlowingCode/DevelopmentConventions/blob/main/code-style.md#punctuation-and-capitalization-rules-for-param-and-return for reference. There're other comments with the same problem in the file, please check.